plugins/utils/alpha.nix

```diff
@@ -139,9 +146,24 @@ in
     let
       layoutDefined = cfg.layout != [ ];
       themeDefined = cfg.theme != null;
+
+      opt = options.plugins.alpha;
     in
     mkIf cfg.enable {
-      extraPlugins = [ cfg.package ] ++ (optional cfg.iconsEnabled pkgs.vimPlugins.nvim-web-devicons);
+      # TODO: deprecated 2024-08-29 remove after 24.11
+      warnings = lib.mkIf opt.iconsEnabled.isDefined [
+        ''
+          nixvim (plugins.alpha):
+          The option definition `plugins.alpha.iconsEnabled' in ${showFiles opt.iconsEnabled.files} has been deprecated; please remove it.
+          You should use `plugins.alpha.iconsPackage' instead.
+        ''
+      ];
+
+      extraPlugins =
+        [ cfg.package ]
+        ++ lib.optional (
+          cfg.iconsPackage != null && (opt.iconsEnabled.isDefined && cfg.iconsEnabled)
```

Comment:
This logic seems wrong; if `iconsEnabled` is not defined then the package will never be added (`cfg.iconsPackage != null && false` is false).

I think you want:

```nix
cfg.iconsPackage != null && (opt.iconsEnabled.isDefined -> cfg.iconsEnabled)
```